doc/releases/release-notes-2.1.rst

```diff
@@ -46,7 +46,49 @@ Bluetooth
 Build and Infrastructure
 ************************
 
-* TBD
+* Various parts of the binding format have been simplified. The format is
```

Comment:
IMO this is too verbose in the release notes, I would just have a short mention and add this to the device tree documentation in some new section.

Reply:
Moved most of it to a separate *Legacy binding syntax* section in the devicetree documentation and added a link from the release notes.

Reply:
@nashif 
Ping